src/datatypes.jl

```diff
@@ -24,7 +24,11 @@ end
 MPIPtr(x::Cint) where T = reinterpret(MPIPtr, x)
 
 
-MPIPtr(x::MPIBuffertype{T}) where {T} = reinterpret(MPIPtr, Base.unsafe_convert(Ptr{T}, x))
+function MPIPtr(x::MPIBuffertype{T}) where {T}
+    GC.@preserve x begin
```

Comment:
the GC.preserve won't help. Why are these methods needed?

Reply:
Sorry for the confusing commit message. Those methods were removed in the next commit (https://github.com/JuliaParallel/MPI.jl/pull/286/commits/0349b34303ffe78bf0beff626c65a7f3bba11a05), not https://github.com/JuliaParallel/MPI.jl/pull/286/commits/0ec5fb0c7bd125d4925654049ea8c7631171f7ac .  I figured that GC won't help so just removed it, but forgot to stage it for https://github.com/JuliaParallel/MPI.jl/pull/286/commits/0ec5fb0c7bd125d4925654049ea8c7631171f7ac.